src/Omnibar.tsx

```diff
@@ -32,15 +32,14 @@ export default class Omnibar<T> extends React.PureComponent<
     }
 
     query = (value: string) => {
-        if (this.props.extensions.length) {
-            search<T>(value, this.props.extensions).then(items => {
-                if (items.length) {
-                    let results = items;
-                    if (this.props.maxResults) {
-                        results = results.slice(0, this.props.maxResults);
-                    }
-                    this.setState({ results, displayResults: true });
-                }
+        if (this.props.extensions.length > 0) {
```

Comment:
I like ensuring that this is a boolean, not rely on implicit validation by the statement

Reply:
good case